src/main/java/de/zalando/aruha/nakadi/repository/ZkSubscriptionRepository.java

```diff
@@ -0,0 +1,142 @@
+package de.zalando.aruha.nakadi.repository;
+
+import de.zalando.aruha.nakadi.NakadiRuntimeException;
+import de.zalando.aruha.nakadi.domain.Cursor;
+import de.zalando.aruha.nakadi.domain.Topology;
+import de.zalando.aruha.nakadi.repository.zookeeper.ZooKeeperHolder;
+import org.apache.zookeeper.CreateMode;
+
+import java.util.List;
+import java.util.UUID;
+
+
+public class ZkSubscriptionRepository implements SubscriptionRepository {
+
+    private ZooKeeperHolder zooKeeperHolder;
+
+    public ZkSubscriptionRepository(final ZooKeeperHolder zooKeeperHolder) {
+        this.zooKeeperHolder = zooKeeperHolder;
+    }
+
+    @Override
+    public void createSubscription(final String subscriptionId, final List<String> topics, final List<Cursor> cursors) {
+        try {
+            createPersistentEmptyNode("/nakadi/subscriptions/%s", subscriptionId);
+            createPersistentEmptyNode("/nakadi/subscriptions/%s/topology", subscriptionId);
+            for (final String topic : topics) {
+                createPersistentEmptyNode("/nakadi/subscriptions/%s/topics/%s", subscriptionId, topic);
+                createPersistentEmptyNode("/nakadi/subscriptions/%s/topics/%s/partitions", subscriptionId, topic);
+            }
+            for (final Cursor cursor: cursors) {
+                final String path = String.format("/nakadi/subscriptions/%s/topics/%s/partitions/%s", subscriptionId,
+                        cursor.getTopic(), cursor.getPartition());
+                zooKeeperHolder
+                        .get()
+                        .create()
```

Comment:
.creatingParentsIfNeeded() will make unnecessary to create parent folders on lines 24 and [26-29]